R/method-differentialAbundance.R

```diff
@@ -154,7 +166,24 @@ function(data, covariate, groupA, groupB, method = c("Maaslin2", "DESeq2"), verb
     data@sampleMetadata@data[[covariate]] <- assignToBinaryGroups(data@sampleMetadata@data[[covariate]], groupA, NULL)
     comparator <- buildBinaryComparator(covariate, 'groupA', 'groupB')
 
-    return(microbiomeComputations::differentialAbundance(data, comparator = comparator, method = method, verbose = verbose))
+    return(microbiomeComputations::internalDiffAbund(data, comparator = comparator, method = method, verbose = verbose))
+})
+
+#' @rdname differentialAbundance-methods
+#' @aliases differentialAbundance,CollectionWithMetadata,character,character,missingOrNULL-method
+setMethod("differentialAbundance", signature("CollectionWithMetadata", "character", "character", "missingOrNULL"),
+function(data, covariate, groupA, groupB, method = c("Maaslin2", "DESeq2"), verbose = c(TRUE, FALSE)) {
+    method <- veupathUtils::matchArg(method)
+    verbose <- veupathUtils::matchArg(verbose)
+
+    covariateDataType <- class(data@sampleMetadata@data[[covariate]])
+    if (!covariateDataType %in% c("factor", "character")) {
+        stop("Argument 'groupA' must be a function when 'covariate' is not a factor or character")
```

Comment:
Can we be more specific here about how it's `data[[covariate]]` that needs to be a factor or character? Maybe something like "... when the 'covariate' variable of the input data is not of type factor or character"?